packages/components/FocusZone/macos/RCTFocusZone.m

```diff
@@ -18,53 +18,26 @@
 // enumerated in the same row (or column) as the current focused view
 static const CGFloat FocusZoneBuffer = 3;
 
-@implementation RCTFocusZone
-
-static inline CGFloat GetDistanceBetweenPoints(NSPoint point1, NSPoint point2)
-{
-	NSPoint delta = NSMakePoint(point1.x - point2.x, point1.y - point2.y);
-	return sqrt(delta.x * delta.x + delta.y * delta.y);
-}
-
-static inline CGFloat GetDistanceBetweenRects(NSRect rect1, NSRect rect2)
-{
-	// Get the top left corner of the rect, top right in RTL
-	bool isRTL = [[RCTI18nUtil sharedInstance] isRTL];
-
-	CGFloat rect1Offset = isRTL ? rect1.size.width : 0;
-	CGFloat rect2Offset = isRTL ? rect2.size.width : 0;
-	
-	NSPoint rect1Corner = NSMakePoint(rect1.origin.x + rect1Offset , rect1.origin.y);
-	NSPoint rect2Corner = NSMakePoint(rect2.origin.x + rect2Offset , rect2.origin.y);
-
-	return GetDistanceBetweenPoints(rect1Corner, rect2Corner);
-}
-
-static inline CGFloat GetMinDistanceBetweenRectVerticesAndPoint(NSRect rect, NSPoint point)
+BOOL ShouldFocusOnView(NSView *view, BOOL forceFocus)
 {
-	return fmin(
-		fmin(GetDistanceBetweenPoints(point, NSMakePoint(NSMinX(rect), NSMinY(rect))),
-			GetDistanceBetweenPoints(point, NSMakePoint(NSMaxX(rect), NSMaxY(rect)))),
-		fmin(GetDistanceBetweenPoints(point, NSMakePoint(NSMaxX(rect), NSMinY(rect))),
-			GetDistanceBetweenPoints(point, NSMakePoint(NSMinX(rect), NSMaxY(rect))))
-	);
+	return forceFocus ? [view acceptsFirstResponder] : [view canBecomeKeyView];
 }
 
 /// Performs a depth first search looking for the first key view in a parent view's view hierarchy.
 /// This function does not take into account the geometric position of the view.
-static NSView *GetFirstKeyViewWithin(NSView *parentView)
+static NSView *GetFirstKeyViewWithin(NSView *parentView, BOOL forceFocus)
 {
-    if ([[parentView subviews] count] < 1)
-    {
-        return nil;
-    }
+		if ([[parentView subviews] count] < 1)
+		{
+				return nil;
+		}
```

Comment:
These whitespace changes look a little odd in Github, just wanted to check that they're actually fine on your side

Reply:
Github defaults to 8 spaces for a tab, instead of 4. I think there's a file we can check in to the repo to fix this. We should also just have... clang format in our repo because I literally get this comment every time I edit native code...

Reply:
Looking locally, it looks fine to me. RCTFocusZone uses Tabs everywhere. I think React Native Core / macOS actually use spaces, so maybe when we add a clang format, we could copy that over and bulk edit all our native code.